monitor/src/test/java/edu/asu/diging/monitor/core/service/impl/AppManagerTest.java

```diff
@@ -60,16 +64,51 @@ public void setUp() {
     }
     
     @Test
-    public void test_addApp_success() throws UnstorableObjectException {
+    public void test_addApp_with_encryption_success() throws UnstorableObjectException {
         String id = "ID1";
         Mockito.when(dbConnection.generateId()).thenReturn(id);
         IApp app = new App();
-        managerToTest.addApp(app);
-        
+        AppForm appForm = new AppForm();
+        appForm.setUsername("user");
+        appForm.setPassword("password");
+        managerToTest.addApp(app, appForm);
+        Mockito.verify(passwordEncryptor).encrypt("password");
         Mockito.verify(dbConnection).store(app);
         Assert.assertEquals(id, app.getId());
     }
     
+    @Test
+    public void test_addApp_without_encryption_success() throws UnstorableObjectException {
+        String id = "ID1";
+        Mockito.when(dbConnection.generateId()).thenReturn(id);
+        IApp app = new App();
+        AppForm appForm = new AppForm();
+        appForm.setUsername("");
+        appForm.setPassword("");
+        managerToTest.addApp(app, appForm);
+        Mockito.verify(passwordEncryptor, never()).encrypt("password");
+        Mockito.verify(dbConnection).store(app);
+        Assert.assertEquals(id, app.getId());
+    }
+    @Test
+    public void test_updateApp_with_encryption_success() {
+        AppForm appForm = new AppForm();
+        appForm.setUsername("user");
+        appForm.setPassword("password");
+        managerToTest.updateApp(app1, appForm);
+        Mockito.verify(passwordEncryptor).encrypt("password");
+        Mockito.verify(dbConnection).update(app1);
+    }
+    
+    @Test
+    public void test_updateApp_without_encryption_success() {
+        AppForm appForm = new AppForm();
+        appForm.setUsername("");
+        appForm.setPassword("");
+        managerToTest.updateApp(app1, appForm);
+        Mockito.verify(passwordEncryptor, never()).encrypt("password");
+        Mockito.verify(dbConnection).update(app1);
```

Comment:
you want to make sure in your update test cases that the app is correctly updated. right now if the method would simply store the app without setting any new values, your test would still pass